packages/nimble/inst/tests/test-mcmc.R

```diff
@@ -2,25 +2,26 @@ source(system.file(file.path('tests', 'test_utils.R'), package = 'nimble'))
 
 context("Testing of default MCMC")
 
-### TODO: add in the special cases for dipper
-
-if(FALSE) { # template for running JAGS for comparison
-  require(R2jags)
-    dir = system.file(file.path('classic-bugs', 'vol2', 'air'), package = 'nimble')
-  data = new.env(); inits = new.env()
-  source(file.path(dir, 'air-data.R'), data)
-  source(file.path(dir, 'air-init.R'), inits)
-  data = as.list(data)
-  inits = list(as.list(inits))
-  out1 <- jags(data = data, inits = inits,
-               parameters.to.save = c('X','theta'), n.chains = 1,
-               n.iter = 100000, n.burnin = 50000, n.thin = 1, model.file = file.path(dir, 'air.bug'),
-               DIC = FALSE, jags.seed = 0)
-  out <- as.mcmc(out1)
-}
-
-
-### Beginning of actual tests
+RwarnLevel <- options('warn')$warn
+options(warn = -1)
+nimbleVerboseSetting <- nimbleOptions('verbose')
+
+## If you do *not* want to write to results files
+##    comment out the sink() call below.  And consider setting verbose = FALSE 
+## To record a new gold file, nimbleOptions('generateGoldFileForMCMCtesting') should contain the path to the directory where you want to put it
+## e.g. nimbleOptions(generateGoldFileForMCMCtesting = getwd())
+## Comparison to the gold file won't work until it is installed with the package.
+nimbleOptions(verbose = TRUE)
+goldFileName <- 'mcmcTestLog_Correct.Rout'
+tempFileName <- 'mcmcTestLog.Rout'
+generatingGoldFile <- !is.null(nimbleOptions('generateGoldFileForMCMCtesting'))
+outputFile <- if(generatingGoldFile) file.path(nimbleOptions('generateGoldFileForMCMCtesting'), goldFileName) else tempFileName
+
+sink(outputFile)
```

Comment:
How did you end up dealing with the travis-fail-on-silence issue? Ideally we could use an R function like the unix `tee` command, so that results were written to both console and a file.